presto-spark/src/main/java/com/facebook/presto/spark/planner/SparkRddPlanner.java

```diff
@@ -0,0 +1,254 @@
+
+                PlanFragment childFragment = childSubPlan.getFragment();
+                RemoteSourceNode remoteSource = getOnlyElement(remoteSources);
+                List<PlanFragmentId> sourceFragmentIds = remoteSource.getSourceFragmentIds();
+                checkArgument(sourceFragmentIds.size() == 1, "expected to have exactly only a single source fragment");
+                checkArgument(childFragment.getId().equals(getOnlyElement(sourceFragmentIds)));
+
+                SparkTaskDescriptor sparkTaskDescriptor = createTaskDescriptor(fragment, ImmutableList.of());
+                byte[] serializedTaskDescriptor = sparkTaskRequestJsonCodec.toJsonBytes(sparkTaskDescriptor);
+
+                PartitioningHandle partitioning = fragment.getPartitioning();
+                if (partitioning.equals(COORDINATOR_DISTRIBUTION)) {
+                    // TODO: We assume COORDINATOR_DISTRIBUTION always means OutputNode
+                    // But it could also be TableFinishNode, in that case we should do collect and table commit on coordinator.
+
+                    // TODO: Do we want to return an RDD for root stage? -- or we should consider the result will not be large?
+                    List<Tuple2<Integer, byte[]>> collect = childRdd.collect();
+                    List<Tuple2<Integer, byte[]>> result = ImmutableList.copyOf(compilerFactory.create().compile(
+                            0,
+                            serializedTaskDescriptor,
+                            ImmutableMap.of(remoteSource.getId().toString(), collect.iterator()),
+                            taskStatsCollector));
+                    return sparkContext.parallelizePairs(result);
+                }
+                else if (partitioning.equals(FIXED_HASH_DISTRIBUTION) ||
+                        // when single distribution - there will be a single partition 0
+                        partitioning.equals(SINGLE_DISTRIBUTION)) {
+                    String planNodeId = remoteSource.getId().toString();
+                    return childRdd
+                            // TODO: What's the difference of using
+                            //  partitionBy/mapPartitionsToPair vs. groupBy vs. mapToPair ???
+                            .partitionBy(partitioning.equals(FIXED_HASH_DISTRIBUTION) ? new HashPartitioner(hashPartitionCount) : new HashPartitioner(1))
+                            .mapPartitionsToPair(createTaskProcessor(compilerFactory, serializedTaskDescriptor, planNodeId, taskStatsCollector));
+                }
+                else {
+                    // SOURCE_DISTRIBUTION || FIXED_PASSTHROUGH_DISTRIBUTION || ARBITRARY_DISTRIBUTION || SCALED_WRITER_DISTRIBUTION || FIXED_BROADCAST_DISTRIBUTION || FIXED_ARBITRARY_DISTRIBUTION
+                    throw new IllegalArgumentException("Unsupported fragment partitioning: " + partitioning);
+                }
+            }
+            else if (children.size() == 2) {
+                // handle simple two-way join
```

Comment:
Either adding a TODO stating this  part requires refactor, or remove them and add support for join later ? :)